binding/Binding/SKTypeface.cs

```diff
@@ -16,6 +17,7 @@ public class SKTypeface : SKObject
 		internal SKTypeface (IntPtr handle)
 			: base (handle)
 		{
+			if (handle == IntPtr.Zero) throw new Exception("Invalid typeface");
```

Comment:
Hi @agallero,

I get you. However, in this case this is an internal constructor (probably should be private) and should not be called outside the class. Also, the reflection-based instantiation might pass a `IntPtr.Zero` value during construction.

The correct way to instantiate this class is by overloading the constructor and passing a specific `IntPtr` and checking *that* value is not `Zero`. The other way is to use `GetObject<T>()` (which should be the only way to construct objects, but this cannot be done in constructors):

```csharp
// these are typical, but should only be used if we know for sure we are 
// constructing a new native object.
// because we are constructing a new object, we check for IntPtr.Zero
public SKMyType()
    : this(SkiaApi.create_my_type())
{
    if (Handle == IntPtr.Zero)
        throw new Exception("unable to create");
}
public SKMyType(int offset)
    : this(IntPtr.Zero) // valid case, possibly by a derived type
{
    var items = SkiaApi.get_some_array();
    if (items == IntPtr.Zero)
        throw new Exception("unable to create");
    Handle = SkiaApi.get_some_value(items, offset);
    if (Handle == IntPtr.Zero)
        throw new Exception("unable to create");
}

// for all other cases, we must go through GetObject so that it can be registered in case
// this method returns the same unmanaged instance on multiple calls.
// we do not want to create multiple managed instances for a single unmanaged instance.
// also, the unmanaged might be NULL, so we either have to do a specific null-check
// for each call, or we just use GetObject
public SKMyType GetMyType()
{
    return GetObject<SKMyType>(...);
}
```

I already made the changes in master to use `GetObject<T>()` (the omission was a bug): https://github.com/mono/SkiaSharp/blob/master/binding/Binding/SKTypeface.cs#L34